web/src/routes/(user)/utilities/[duplicates]/[[photos=photos]]/[[assetId=id]]/+page.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import UserPageLayout from '$lib/components/layouts/user-page-layout.svelte';
+  import DuplicatesCompareControl from '$lib/components/utilities-page/duplicates/duplicates-compare-control.svelte';
+  import { resolveDuplicates } from '@immich/sdk';
+
+  import type { PageData } from './$types';
+  import { handleError } from '$lib/utils/handle-error';
+  import {
+    NotificationType,
+    notificationController,
+  } from '$lib/components/shared-components/notification/notification';
+
+  export let data: PageData;
+
+  const handleOnResolve = async (duplicateId: string, trashIds: string[]) => {
+    try {
+      await resolveDuplicates({ resolveDuplicatesDto: { duplicateId, ids: trashIds } });
+      data.duplicates = data.duplicates.filter((duplicate) => duplicate.duplicateId !== duplicateId);
+
+      if (trashIds.length === 0) {
+        return;
+      }
+
+      notificationController.show({
+        message: `Moved ${trashIds.length} to trash`,
+        type: NotificationType.Info,
+      });
+    } catch (error) {
+      handleError(error, 'Unable to resolve duplicate');
+    }
+  };
+</script>
+
+<UserPageLayout title={data.meta.title + ` (${data.duplicates.length})`} scrollbar={true}>
+  <div class="mt-4">
+    {#if data.duplicates && data.duplicates.length > 0}
+      <div class="mb-4 text-sm dark:text-white">
+        <p>Resolve the following duplication by keeping or moving assets to the trash.</p>
+      </div>
+      {#key data.duplicates[0].duplicateId}
```

Comment:
I trust that it's needed but I'd like to understand why we need a key here? I thought you generally only need keys when looping over a dataset/creating many components.

Reply:
If you don't have the key, Svelte doesn't know when the nested component change to properly rerender

Reply:
Ah I see, thanks!